api/envoy/extensions/upstreams/http/v3/http_protocol_options.proto

```diff
@@ -75,6 +76,20 @@ message HttpProtocolOptions {
     config.core.v3.Http2ProtocolOptions http2_protocol_options = 2;
   }
 
+  // If this is used, the cluster can will use both HTTP/1 and HTTP/2, whichever
+  // protocol is negotiated by ALPN with the upstream.
+  // Clusters configured with *AutoHttpConfig* will use the highest available
+  // protocol; HTTP/2 if supported, otherwise HTTP/1.
+  // If the upstream does not support ALPN, *AutoHttpConfig* will fail over to HTTP/1.
+  // This can only be used with transport sockets which support ALPN. The
```

Comment:
Clarify that this will cause an error? It's a bit unclear when read with the previous section on falling back to H1?